### PR TITLE
Release v0.36.0

### PR DESCRIPTION
**Pull request by jcheng-splunk**

### CODE REVIEW

File: cmd/otelcol/config/collector/agent_config.yaml [hunk 1]
```diff
@@ -142,7 +142,8 @@ exporters:
   # Send to gateway
   otlp:
     endpoint: "${SPLUNK_GATEWAY_URL}:4317"
-    insecure: true
+    tls:
+      insecure: true
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/agent_config.yaml:146]: I believe this is something we need to worry about. 

If user upgrades linux package collector likely won't be able to start. Is this desired behavior?

If not we can probably add another case in`internal/configconverter` to handle that?

cc @emaderer @flands  @jcheng-splunk @rmfitzpatrick

Comment (dmitryax) [reply cmd/otelcol/config/collector/agent_config.yaml:146]: Just confirmed: collector fails to start with the old config even if the `otlp` exported is not added to the pipeline by default

File: internal/configconverter/otlp_insecure_mover.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	out := configparser.NewConfigMap()
+	for _, k := range in.AllKeys() {
+		v := in.Get(k)
+		match := insecureRE.FindStringSubmatch(k)
+		if match == nil {
+			out.Set(k, v)
+		} else {
+			tlsKey := fmt.Sprintf("exporters::otlp%s::tls::insecure", match[1])
+			log.Printf("Unsupported key found: %s. Moving to %s\n", k, tlsKey)
+			out.Set(tlsKey, v)
```
Comment (dmitryax) [inline internal/configconverter/otlp_insecure_mover.go:37]: This is awesome!

File: cmd/otelcol/config/collector/agent_config.yaml (hunk 1 above)
Comment (pmcollins) [reply cmd/otelcol/config/collector/agent_config.yaml:146]: Hi @dmitryax, I just pushed a change to move that config key.

**Review by dmitryax — APPROVED** [review]

File: cmd/otelcol/config/collector/agent_config.yaml (hunk 1 above)
Comment (dmitryax) [reply cmd/otelcol/config/collector/agent_config.yaml:146]: Thanks @pmcollins !

File: internal/configconverter/otlp_insecure_mover.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	insecureRE, _ := regexp.Compile(expr)
+	out := configparser.NewConfigMap()
+	for _, k := range in.AllKeys() {
+		v := in.Get(k)
+		match := insecureRE.FindStringSubmatch(k)
+		if match == nil {
+			out.Set(k, v)
+		} else {
+			tlsKey := fmt.Sprintf("exporters::otlp%s::tls::insecure", match[1])
+			log.Printf("Unsupported key found: %s. Moving to %s\n", k, tlsKey)
```
Comment (dmitryax) [inline internal/configconverter/otlp_insecure_mover.go:36]: Maybe we should change the wording to encourage user to update the config so we can drop it later on?

Something like:
```
[DEPRECATION NOTICE] Configuration parameter "exporters -> otlp -> insecure" is deprecated. 
Please move it to "exporters -> otlp -> tls -> insecure".
More details: https://github.com/open-telemetry/opentelemetry-collector/pull/4063/
```

@emaderer thoughts?

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,20 @@
...
 ## Unreleased
 
+## v0.36.0
+
+This Splunk OpenTelemetry Connector release includes changes from the [opentelemetry-collector v0.36.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.36.0) and the [opentelemetry-collector-contrib v0.36.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.36.0) releases.
+
+### 💡 Enhancements 💡
+
+- Handle backwards compatibility of internal ballast removal (#759)
+- Update bundled Smart Agent to [v5.14.0](https://github.com/signalfx/signalfx-agent/releases/tag/v5.14.0)
```
Comment (rmfitzpatrick) [inline CHANGELOG.md:12]:
```suggestion
- Update bundled Smart Agent to [v5.14.0](https://github.com/signalfx/signalfx-agent/releases/tag/v5.14.0)
- Automatically relocate removed OTLP exporter "insecure" field (#783)
```

**Pull request merged**